src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -534,6 +611,27 @@ public ICommand BrowseDirectoryCommand
             }
         }
 
+        public ICommand FindRemoteMachineCommand
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref _findRemoteMachineCommand, () =>
+                new DelegateCommand(state =>
+                {
+                    SVsServiceProvider serviceProvider = Project.Services.ExportProvider.GetExportedValue<SVsServiceProvider>();
+
+                    // HACK: Using VsUIService directly here avoids using a banned UI (serviceProvider.GetService) and can be replaced easily when MEF is hooked up to this page
+                    var vsService = new VsUIService<SVsDebugRemoteDiscoveryUI, IVsDebugRemoteDiscoveryUI>(serviceProvider, _projectThreadingService.JoinableTaskContext.Context);
+
+                    if (ErrorHandler.Succeeded(vsService.Value.SelectRemoteInstanceViaDlg(RemoteDebugMachine, RemoteAuthenticationProvider.PortSupplier, (uint)DEBUG_REMOTE_DISCOVERY_FLAGS.DRD_NONE, out string remoteMachine, out Guid portSupplier)))
```

Comment:
This needs a little more thought as "DRD_SUPPORTS_UNIVERSAL_AUTH" needs to be passed if Universal is actually selected.